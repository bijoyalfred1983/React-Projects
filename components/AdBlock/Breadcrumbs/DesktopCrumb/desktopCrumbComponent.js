/* @flow */
import React, { PureComponent } from 'react';
import { Breadcrumb } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import * as styles from '../styles.scss';
import { analyticsBreadCrumb } from '../../../containers/Analytics/DataLayer';
import { jsTrack } from '../../../containers/Analytics/ClickTrack';

type Props = {
  data: any
};

class DesktopCrumb extends PureComponent<Props> {
  handleClick = (e: any, linkName: any) => {
    jsTrack.breadCrumbInteraction(linkName);
  };
  render() {
    const breadcrumbsList = this.props.data;
    analyticsBreadCrumb({
      pageHierarchy: `${this.props.data
        .map(item => item.displayName)
        .join('|')}`
    });
    return (
      <Breadcrumb className={styles.breadcrumbLinkAZ}>
        {breadcrumbsList &&
          breadcrumbsList.map((breadcrumbItem, index) => {
            if (
              parseInt(index, 10) === parseInt(breadcrumbsList.length - 1, 10)
            ) {
              return (
                <Breadcrumb.Section
                  active
                  key={breadcrumbItem.displayName}
                  className={styles.activeLink}
                >
                  {breadcrumbItem.displayName}
                </Breadcrumb.Section>
              );
            }
            return (
              <span key={breadcrumbItem.displayName}>
                <Breadcrumb.Section className={styles.homeLink}>
                  <Link
                    to={{
                      pathname: breadcrumbItem.url || `/`,
                      state: { pageType: breadcrumbItem.pageType }
                    }}
                    className={styles.homeLink}
                    id="breadcrumbItem"
                    onClick={e =>
                      this.handleClick(e, breadcrumbItem.displayName)
                    }
                  >
                    {breadcrumbItem.displayName}
                  </Link>
                </Breadcrumb.Section>
                <Breadcrumb.Divider className={styles.dividerLink} />
              </span>
            );
          })}
      </Breadcrumb>
    );
  }
}

export default DesktopCrumb;
