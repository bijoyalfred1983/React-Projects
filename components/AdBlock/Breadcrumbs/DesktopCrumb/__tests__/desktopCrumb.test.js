import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import DesktopCrumb from '../desktopCrumbComponent';

describe('<DesktopCrumb />', () => {
  let PROPS;
  beforeEach(() => {
    PROPS = {
      data: [
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Parts',
          url: '/Parts'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Brakes And Traction Control',
          url: '/parts/brakes-and-traction-control'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Disc Brake System',
          url: '/parts/brakes-and-traction-control/disc-brake-system'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Brake Pads',
          url: '/brakes-and-traction-control/brake-pads'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Duralast PlatinumBattery_BT',
          url: ''
        }
      ]
    };
  });

  test('renders on desktop correctly', () => {
    const wrapper = mount(<DesktopCrumb {...PROPS} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('onClick renders correctly', () => {
    const wrapper = shallow(<DesktopCrumb {...PROPS} />);
     wrapper.find('#breadcrumbItem').at(0).simulate('click');
     expect(toJson(wrapper)).toMatchSnapshot();

   });

});
