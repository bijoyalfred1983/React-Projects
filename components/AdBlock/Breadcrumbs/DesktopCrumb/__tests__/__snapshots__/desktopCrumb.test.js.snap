// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DesktopCrumb /> onClick renders correctly 1`] = `
<Breadcrumb>
  <span
    key="Parts"
  >
    <BreadcrumbSection>
      <Link
        id="breadcrumbItem"
        onClick={[Function]}
        replace={false}
        to={
          Object {
            "pathname": "/Parts",
            "state": Object {
              "pageType": "CategoryGrouping",
            },
          }
        }
      >
        Parts
      </Link>
    </BreadcrumbSection>
    <BreadcrumbDivider />
  </span>
  <span
    key="Brakes And Traction Control"
  >
    <BreadcrumbSection>
      <Link
        id="breadcrumbItem"
        onClick={[Function]}
        replace={false}
        to={
          Object {
            "pathname": "/parts/brakes-and-traction-control",
            "state": Object {
              "pageType": "CategoryGrouping",
            },
          }
        }
      >
        Brakes And Traction Control
      </Link>
    </BreadcrumbSection>
    <BreadcrumbDivider />
  </span>
  <span
    key="Disc Brake System"
  >
    <BreadcrumbSection>
      <Link
        id="breadcrumbItem"
        onClick={[Function]}
        replace={false}
        to={
          Object {
            "pathname": "/parts/brakes-and-traction-control/disc-brake-system",
            "state": Object {
              "pageType": "CategoryGrouping",
            },
          }
        }
      >
        Disc Brake System
      </Link>
    </BreadcrumbSection>
    <BreadcrumbDivider />
  </span>
  <span
    key="Brake Pads"
  >
    <BreadcrumbSection>
      <Link
        id="breadcrumbItem"
        onClick={[Function]}
        replace={false}
        to={
          Object {
            "pathname": "/brakes-and-traction-control/brake-pads",
            "state": Object {
              "pageType": "CategoryGrouping",
            },
          }
        }
      >
        Brake Pads
      </Link>
    </BreadcrumbSection>
    <BreadcrumbDivider />
  </span>
  <BreadcrumbSection
    active={true}
    key="Duralast PlatinumBattery_BT"
  >
    Duralast PlatinumBattery_BT
  </BreadcrumbSection>
</Breadcrumb>
`;
